utils/checks.py

```diff
@@ -1,13 +1,15 @@
-import discord
 from discord.ext import commands
+from db.models import GuildData
 
 def cogify(check):
     return lambda self, ctx: check.predicate(ctx)
 
 
-def is_moderator(ctx: commands.Context):
-    moderator_role_id = None  # TODO: retrieve moderator role ID
-    return any(role.id == moderator_role_id for role in ctx.author.roles)
+async def is_moderator(ctx: commands.Context):
+    assert await commands.guild_only().predicate(ctx)
+    guild_data = await GuildData.get_or_none(guild_id=ctx.guild.id)
+    moderator_role_id = guild_data.moderator_id if guild_data else None
+    return any(role.id == moderator_role_id for role in ctx.author.roles) if moderator_role_id else False
```

Comment:
Is the ternary function here explicitly needed? I will in theory be faster, if there is no moderator role, however, that is essentially never. And if there is no moderator role, it'll just be None, but then the any check will always fail, as none of the role ids will be None.

Reply:
It was originally added so that we skip going through all the roles and just return early if there is no moderator role (so yeah a performance related addition). So no, it's not explicitly needed and will continue to return false if the moderator role is `None` (was considered before). Can remove it 👍

Reply:
> Actually, I suggested making the `moderator_role_id == None` case distinct for the sake of error messaging; I have this in a local branch:
> 
> ```python
> class GuildDataMissing(commands.CommandError):
>     pass
> ...
> async def _is_moderator(ctx: commands.Context):
>     ...
>     if not moderator_role_id:
>         raise GuildDataMissing('Moderator role not set for this guild.')
>     ...
> ```
> 
> It's more descriptive than discord.py's "The check functions for command update failed.", but I am not sure how necessary it is.

For this reason, I'm going to keep this in the codebase, which can be changed later when cog checks are changed again. How does this sound @raynichc ?

Reply:
I think that should be okay. Slight nitpick with the comment though, it should be `moderator_role_id is None`, though in the code I think the `not moderator_role_id` is good.